java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -259,12 +271,72 @@ public MapFunction getTransformImpl(Transform transform) {
             f = getFilterTokenByRegex((Transform.FilterTokenByRegex) transform);
         } else if (transform instanceof Transform.Tokenize) {
             f = getTokenize((Transform.Tokenize) transform);
+        } else if (transform instanceof HashIp) {
+            f = getHashIp((HashIp) transform);
+        } else if (transform instanceof EncryptIp) {
+            f = getEncryptIp((EncryptIp) transform);
         } else {
             throw new IllegalArgumentException("Unknown transform type: " + transform.getClass().getName());
         }
         return f;
     }
 
+    @VisibleForTesting
+    MapFunction getEncryptIp(EncryptIp transform) {
+        return (s, jsonConfiguration) -> {
+            if (!(s instanceof String)) {
+                if (s != null) {
+                    log.warning("value matched by " + transform + " not of type String");
+                }
+                return null;
+            } else if (StringUtils.isBlank((String) s)) {
+                return s;
+            } else {
+                try {
+                    // Parse the IP address string and convert back to string to get the canonical form.
+                    InetAddress address = InetAddress.getByName((String) s);
+                    // already canonicalized, so can pass identity
+                    return urlSafePseudonymEncoder.encode(Pseudonym.builder()
+                            .hash(ipHashStrategy.getToken(address.getHostAddress()))
+                            .reversible(ipEncryptStrategy.getReversibleToken(address.getHostAddress()))
```

Comment:
why do we want to reverse ips on our side??

Reply:
we don't reverse IPs on our side. 

We support `encryptIP` and `hashIP` transforms; customers can choose. I think use-case for the encryption would be that any IP that customer sees associated with above some number of people can be safely reversed/decrypted, as it's not identifying the home of a person. 

Customers may have use-case for having plain IPs to be able to localize the physical office / building / access point in such situations. With encryption, they could build a list of IPs by number of associated people, and decrypt only those that they're OK with from privacy perspective.